packages/app-desktop/gui/UpdateNotification/UpdateNotification.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import { useCallback, useContext, useEffect, useMemo, useRef } from 'react';
+import { themeStyle } from '@joplin/lib/theme';
+import NotyfContext from '../NotyfContext';
+import { UpdateInfo } from 'electron-updater';
+import { ipcRenderer, IpcRendererEvent } from 'electron';
+import { AutoUpdaterEvents } from '../../services/autoUpdater/AutoUpdaterService';
+import { NotyfNotification } from 'notyf';
+import { _ } from '@joplin/lib/locale';
+
+interface UpdateNotificationProps {
+	themeId: number;
+}
+
+export enum UpdateNotificationEvents {
+	ApplyUpdate = 'apply-update',
+	Dismiss = 'dismiss-update-notification',
+}
+
+const changelogLink = 'https://joplinapp.org/help/about/changelog/desktop/';
```

Comment:
For the changelog it's a bit tricky because it is updated with a delay, so it may not have the information by the time the user gets the notification. So to be sure let's link instead to https://github.com/laurent22/joplin/releases